Mathlib/NumberTheory/EulerProduct/ExpLog.lean

```diff
@@ -0,0 +1,52 @@
+/-
+Copyright (c) 2024 Michael Stoll. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Michael Stoll
+-/
+import Mathlib.Analysis.SpecialFunctions.Complex.LogDeriv
+import Mathlib.NumberTheory.EulerProduct.Basic
+
+/-!
+# Logarithms of Euler Products
+
+We consider `f : ℕ →*₀ ℂ` and show that `exp (∑ p in Primes, log (1 - f p)⁻¹) = ∑ n : ℕ, f n`
+under suitable conditions on `f`. This can be seen as a logarithmic version of the
+Euler product for `f`.
+-/
+
+open Complex
+
+attribute [fun_prop] DifferentiableAt.clog
```

Comment:
Any reason this shouldn't be on the declaration?

Reply:
Probably not, except that `fun_prop` is not able to prove the side condition by itself (see the `fun_prop (disch := assumption)` in the proof below.

Reply:
I assume that *if* I put the attribute directly on the declaration, I should add it to *all* the declarations in `Mathlib.Analysis.SpecialFunctions.Complex.LogDeriv`?

Reply:
For now, I have made the attribute `local`, to avoid downstream effects.